### PR TITLE
[MINVOKER-313] Get rid of maven-artifact-transfer - InstallMojo refactor

### PR DESCRIPTION
**Pull request by slawekjaranowski**
- refactor InstallMojo to use Resolver API
- add possibility to configure scope for installing artifacts
- remove maven-artifact-transfer and maven-common-artifact-filters from dependencies

---

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MINVOKER) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MINVOKER-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MINVOKER-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: src/main/java/org/apache/maven/plugins/invoker/MetadataUtils.java [hunk 1]
```diff
@@ -1,146 +0,0 @@
...
-import org.codehaus.plexus.util.xml.Xpp3DomUtils;
-import org.codehaus.plexus.util.xml.Xpp3DomWriter;
-import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
-
-/**
- * Provides utility methods for artifact metadata processing.
- *
- * @author Benjamin Bentmann
- */
-class MetadataUtils {
```
Comment (cstamas) [inline src/main/java/org/apache/maven/plugins/invoker/MetadataUtils.java:46]: This, cool!
Solves https://issues.apache.org/jira/browse/MRESOLVER-224

Comment (cstamas) [reply src/main/java/org/apache/maven/plugins/invoker/MetadataUtils.java:46]: and https://issues.apache.org/jira/browse/MNG-7548

**Review by cstamas — APPROVED** [review]

**Review by gnodet — APPROVED** [review]

File: src/main/java/org/apache/maven/plugins/invoker/MetadataUtils.java (hunk 1 above)
Comment (slawekjaranowski) [reply src/main/java/org/apache/maven/plugins/invoker/MetadataUtils.java:46]: issues linked 😄

**Comment by jira-importer** [comment]
Resolve #551

**Pull request merged**